stripe/http_client.py

```diff
@@ -102,6 +102,10 @@ def __init__(self, timeout=80, session=None, **kwargs):
         self._timeout = timeout
         self._session = session or requests.Session()
 
+    def __del__(self):
+        if self._session:
+            self._session.close()
```

Comment:
Under Python 3, requests will sometimes return `ResourceWarning`s after execution (cf. https://github.com/requests/requests/issues/1882). There are some tests where we explicitly catch and count warnings (to check that we're raising `DeprecationWarning`s) and this bug/feature was interfering with the tests.

It's apparently not great practice to close resources in destructors, but it does fix the tests and I suppose it's better than doing nothing at all.

This should not be related to pytest though and I'm not entirely sure why we never ran into this issue before.

Reply:
Not sure either, but certainly a problem :$ This looks strictly better than what we had before. Unfortunately the "real" solution is probably to implement our own `close` method for users to call into manually.